### PR TITLE
`unittest.TestCase.subTest` support

### LINKED ISSUE #91: Suboptimal support for "unittest.TestCase.subTest": missing exception/traceback information
```
from unittest import TestCase

class Tests(TestCase):
  def test_subTest(self):
    with self.subTest("subtest 1"):
      self.assertEqual(0, 1)
    with self.subTest("subtest 2"):
      self.assertEqual(0, 1)

  def test_topTest(self):
    self.assertEqual(0, 1)
```
produces the following `zope.testrunner` test result:
```
xmlsec3: PYTHONPATH=. zope-testrunner -v -s package --test-path=.
Running tests at level 1
Running zope.testrunner.layer.UnitTests tests:
  Set up zope.testrunner.layer.UnitTests in 0.000 seconds.
  Running:
..

Failure in test test_topTest (package.tests.Tests)
Traceback (most recent call last):
  File "/usr/lib/python3.6/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib/python3.6/unittest/case.py", line 605, in run
    testMethod()
  File "/home/dieter/tmp/xmlsec3/package/tests.py", line 11, in test_topTest
    self.assertEqual(0, 1)
  File "/usr/lib/python3.6/unittest/case.py", line 829, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/usr/lib/python3.6/unittest/case.py", line 822, in _baseAssertEqual
    raise self.failureException(msg)
AssertionError: 0 != 1


  Ran 2 tests with 3 failures, 0 errors and 0 skipped in 0.002 seconds.
Tearing down left over layers:
  Tear down zope.testrunner.layer.UnitTests in 0.000 seconds.

Tests with failures:
   test_subTest (package.tests.Tests) [subtest 1]
   test_subTest (package.tests.Tests) [subtest 2]
   test_topTest (package.tests.Tests)
```

As you can see, the exception and traceback information is shown for a "noraml" test but unavailable for a "subTest". This makes the `unittest.TestCase.subTest` feature virtually unusable with `zope.testrunner`.

### PR DESCRIPTION
**Pull request by d-maurer**
Fixes #91.

### CODE REVIEW

**Comment by d-maurer** [comment]
@dataflake, @icemac  Do you know a potential cause why (most) GHA checks are not run and what to do about it?

**Comment by dataflake** [comment]
If you click on https://github.com/zopefoundation/zope.testrunner/actions you will see that the "tests" workflow shows as "disabled". Click on it and enable it. Workflows are automatically disabled by GH after a while. This is normal.

**Comment by d-maurer** [comment]
The remaining test failures are unrelated to this PR:
* there are significant changes in Python 3.13 which let various tests fail -- those problems are better handled in their own PR
* under Windows, several tests related to thread support are failing for `PyPy3`. I assume that timing differences are responsible.

**Review by dataflake — APPROVED** [review]
Small oversight: this is missing a py313-subunit test in tox.ini.

I agree that the Python 3.13 failures were there before, tackling them in a separate PR is fine.

**Comment by icemac** [comment]
> If you click on https://github.com/zopefoundation/zope.testrunner/actions you will see that the "tests" workflow shows as "disabled". Click on it and enable it. Workflows are automatically disabled by GH after a while. This is normal.

There is https://github.com/zopefoundation/meta/blob/master/config/re-enable-actions.py which I run from time to time (like just now) which re-enables GHA for all repositories configured via `meta/config`.

File: src/zope/testrunner/runner.py [hunk 1]
```diff
@@ -974,6 +974,29 @@ def addSkip(self, test, reason):
...
+        report(subtest, time.time() - self._start_time,
+               exc_info, stdout=stdout, stderr=stderr)
+
+        unittest.TestResult.addSubTest(self, test, subtest, exc_info)
+
+        if self.options.post_mortem:
+            if self.options.resume_layer:
+                self.options.output.error_with_banner("Can't post-mortem debug"
+                                                      " when running a layer"
+                                                      " as a subprocess!")
```
Comment (icemac) [inline src/zope/testrunner/runner.py:994]:
```suggestion
                self.options.output.error_with_banner(
                    "Can't post-mortem debug when running a layer as"
                    " a subprocess!")
```

**Review by icemac — APPROVED** [review]
Thank you for implementing this feature.

**Review by gforcada — APPROVED** [review]

**Comment by d-maurer** [comment]
The formerly failing and now succeeding `PyPy3` tests for Windows show that we have a race condition related to the thread support. Maybe, it affects Windows only..

**Pull request merged**